pkg/ipcache/ipcache.go

```diff
@@ -54,6 +55,8 @@ type K8sMetadata struct {
 	Namespace string
 	// PodName is the Kubernetes pod name behind the IP
 	PodName string
+	// NamedPorts is the set of named ports for the pod
+	NamedPorts policy.NamedPortsMap
```

Comment:
I would be extremely careful with this. Adding this field removes the ability to compare 2 variables created from this structure type with the `==` operator.

Reply:
It has an `Equal` method already, I did expand it. Could also consider making `NamedPorts` a separate member/argument, although port names are semantically "k8s metadata".